postgresql/provider.go

```diff
@@ -130,17 +136,41 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	versionStr := d.Get("expected_version").(string)
 	version, _ := semver.Parse(versionStr)
 
-	config := Config{
-		Host:              d.Get("host").(string),
-		Port:              d.Get("port").(int),
-		Database:          d.Get("database").(string),
-		Username:          d.Get("username").(string),
-		Password:          d.Get("password").(string),
-		SSLMode:           sslMode,
-		ApplicationName:   tfAppName(),
-		ConnectTimeoutSec: d.Get("connect_timeout").(int),
-		MaxConns:          d.Get("max_connections").(int),
-		ExpectedVersion:   version,
+	var config Config
+	connString := d.Get("connection_string").(string)
+
+	if connString != "" {
+		conn, err := parseConnectionString(connString)
+
```

Comment:
Style nit: no extra whitespace between the above statement and its error handling.